controller_interface/include/controller_interface/controller_interface.hpp

```diff
@@ -34,6 +36,19 @@ enum class return_type : std::uint8_t
   ERROR = 1,
 };
 
+enum class configuration_type : std::uint8_t
```

Comment:
Um... This is super general and I have no clue what'd use this for. Please either document this, give those fields more descriptive names or let's think about alternative solutions.

Reply:
Maybe something like this:
```suggestion
enum class interface_configuration_type : std::uint8_t
```

Reply:
c5f65f9